coffea/btag_tools/btagscalefactor.py

```diff
@@ -212,7 +206,12 @@ def eval(self, systematic, flavor, abseta, pt, discr=None, ignore_missing=False)
                 var = numpy.clip(discr, corr[2][0], corr[2][-1])
             else:
                 var = numpy.clip(pt, corr[1][0], corr[1][-1])
-            func(var, out=out, where=(mapidx == ifunc))
+            where = (mapidx == ifunc)
+            if isinstance(func, float):
+                out[where] = func
+            else:
+                tmp = func(var)
```

Comment:
How about `tmp = func(var[where])` and `out[where] = tmp` ? Saves evaluation on unnecessary inputs

Reply:
sure. I didn't find a speed difference in a test on 1M random inputs, but might be nicer.

Reply:
Ha!  Goes to show how assumptions about performance are often wrong.. turns out we can do extra math about as fast as allocating a smaller array
Feel free to change or leave it if you tested.